packages/api/src/index.ts

```diff
@@ -1,5 +1,54 @@
+import 'cross-fetch/polyfill';
+
+import { API_ENDPOINT } from './constants';
+import { HTTPError } from './httpError';
+import { IApiOptions, IFetchOptions, IGraphQLRequest, IGraphQLResponse, IGraphQLVariables } from './types';
+
 export class Api {
-  constructor() {
-    console.log('api instance has created');
+  private readonly url: string;
+  private readonly headers: IApiOptions['headers'];
+
+  constructor(options: IApiOptions) {
+    const { workspaceId, headers } = options;
+
+    this.url = `${API_ENDPOINT}/${workspaceId}`;
+    this.headers = headers;
+  }
+
+  public async request<T>(
+    query: string,
+    variables?: IGraphQLVariables,
+    options: IFetchOptions = { headers: {} },
+  ): Promise<IGraphQLResponse<T>> {
+    const request: IGraphQLRequest = {
+      query,
```

Comment:
@sebscholl Should we validate queries passed to `request`, `query`, `mutation` methods on sdk side? Or delegate this job to the server?

Server in this case returns HTTP errors, not GraphQL.

Reply:
I'd leave it to server side, which would return a GraphQL error that the method is missing, correct?

Reply:
If a user passes wrong query. I.e. `quuuery { tablesList { items { id } } }` the server will return HTTP error with `Bad Request` status. It seems to me not so obvious. User won't easily understand that the error is in the query by this short message.